### ISSUE TITLE
Parsing error: Declaration expected.

### ISSUE THREAD

**Issue by MSshizuka** [eslint/eslint#16892]
### Environment

Node version: v16.15.1
npm version: 8.11.0
Local ESLint version: 
Global ESLint version: 7.30.0
Operating System: win10

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
    sourceType: 'module',
  },
  plugins: ['@typescript-eslint/eslint-plugin'],
  extends: ['plugin:@typescript-eslint/recommended', 'plugin:prettier/recommended'],
  root: true,
  env: {
    node: true,
    jest: true,
  },
  ignorePatterns: ['.eslintrc.js'],
  rules: {
    '@typescript-eslint/interface-name-prefix': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
  },
};
```
</details>

```js
![Snipaste_2023-02-14_15-53-34](https://user-images.githubusercontent.com/47559134/218678430-5f712dca-0873-4050-8431-cd20114a0dc5.png)

<!-- Paste your code here -->
```
![Snipaste_2023-02-14_15-53-30](https://user-images.githubusercontent.com/47559134/218677873-bc9484d2-069b-41e0-a619-74cdbfb3774f.png)

### What did you expect to happen?

There should be no error

### What actually happened?

is error

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by MSshizuka** [comment]
![Snipaste_2023-02-14_15-53-34](https://user-images.githubusercontent.com/47559134/218678997-f871fa30-49f2-4ba8-9a4d-96d321c70b8d.png)

**Comment by MSshizuka** [comment]
> Hi @MSshizuka, thanks for the issue!
> 
> Since you're using `@typescript-eslint/parser`, for parser-related problems please submit issues on the typescript-eslint repo:
> 
> https://github.com/typescript-eslint/typescript-eslint
> 
> In the examples you provided, I'm guessing that `;` is causing parsing errors.

It really is, thanks a lot!
